tests/components/lunatone/test_light.py

```diff
@@ -0,0 +1,126 @@
+
+from . import setup_integration
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+TEST_ENTITY_ID = "light.device_1"
+
+
+async def test_setup(
+    hass: HomeAssistant,
+    mock_lunatone_devices: AsyncMock,
+    mock_lunatone_info: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    device_registry: dr.DeviceRegistry,
+    entity_registry: er.EntityRegistry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test the Lunatone configuration entry loading/unloading."""
+    await setup_integration(hass, mock_config_entry)
+
+    entities = hass.states.async_all(Platform.LIGHT)
+    for entity_state in entities:
+        entity_entry = entity_registry.async_get(entity_state.entity_id)
+        assert entity_entry == snapshot(name=f"{entity_entry.entity_id}-entry")
+        assert entity_state == snapshot(name=f"{entity_entry.entity_id}-state")
+
+
+async def test_turn_on_off(
+    hass: HomeAssistant,
+    mock_lunatone_devices: AsyncMock,
+    mock_lunatone_info: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test the light can be turned on and off."""
+
+    def fake_control(device: Device, control_data: ControlData):
+        if control_data.switchable is not None:
+            device._data.features.switchable.status = control_data.switchable
+
+    device_list = [
+        Device(mock_lunatone_devices._auth, d)
+        for d in mock_lunatone_devices._data.devices
+    ]
+    for dev in device_list:
+        dev.async_control = AsyncMock(
+            side_effect=lambda data, d=dev: fake_control(d, data)
+        )
+        dev.async_update = AsyncMock()
+
+    with patch.object(Devices, "devices", new_callable=PropertyMock) as mock_prop:
+        mock_prop.return_value = device_list
+
+        await setup_integration(hass, mock_config_entry)
+
+        await hass.services.async_call(
+            LIGHT_DOMAIN,
+            SERVICE_TURN_ON,
+            {ATTR_ENTITY_ID: TEST_ENTITY_ID},
+            blocking=True,
+        )
+
+        assert mock_lunatone_devices.devices[0].is_on
+
+        await hass.services.async_call(
+            LIGHT_DOMAIN,
+            SERVICE_TURN_OFF,
+            {ATTR_ENTITY_ID: TEST_ENTITY_ID},
+            blocking=True,
+        )
+
+        assert not mock_lunatone_devices.devices[0].is_on
+        assert mock_lunatone_devices.devices[0].async_control.call_count == 2
+
+
+async def test_coordinator_update_handling(
+    hass: HomeAssistant,
+    mock_lunatone_devices: AsyncMock,
+    mock_lunatone_info: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test the coordinator update handling."""
+
+    async def fake_update():
+        new_device = mock_lunatone_devices._data.devices[0].model_copy(deep=True)
+        new_device.features.switchable.status = (
+            not new_device.features.switchable.status
+        )
+        mock_lunatone_devices._data.devices[0] = new_device
```

Comment:
```suggestion
        device = mock_lunatone_devices._data.devices[0]
        device.features.switchable.status = (
            not new_device.features.switchable.status
        )
```
can we simplify to this?

Reply:
[f450a9f](https://github.com/home-assistant/core/pull/149182/commits/f450a9f0472410e922f0c6e9f7df5a75f29c7197)